binary_gentoo/internal/cli/tree_diff.py

```diff
@@ -22,11 +37,32 @@ def iterate_new_and_changed_ebuilds(config):
         for ebuild_file in ebuild_files:
             old_portdir_ebuild_file = os.path.join(config.old_portdir, category_plus_package,
                                                    ebuild_file)
+            new_portdir_ebuild_file = os.path.join(root, ebuild_file)
+
+            # don't output if files are identical
             if os.path.exists(old_portdir_ebuild_file):
-                new_portdir_ebuild_file = os.path.join(root, ebuild_file)
                 if filecmp.cmp(old_portdir_ebuild_file, new_portdir_ebuild_file):
                     continue
 
+            if config.keywords:
+                with open(new_portdir_ebuild_file, "r") as ifile:
+                    new_ebuild_content = ifile.read()
+                new_ebuild_relevant_keywords = _get_relevant_keywords_set_for(new_ebuild_content, config.keywords)
```

Comment:
I wonder if extracting this whole condition body to a function some way might make the function `iterate_new_and_changed_ebuilds` quicker to understand.